shell/completion.bash

```diff
@@ -39,10 +39,10 @@ __fzf_comprun() {
     _fzf_comprun "$@"
   elif [[ -n "${TMUX_PANE-}" ]] && { [[ "${FZF_TMUX:-0}" != 0 ]] || [[ -n "${FZF_TMUX_OPTS-}" ]]; }; then
     shift
-    fzf-tmux ${FZF_TMUX_OPTS:--d${FZF_TMUX_HEIGHT:-40%}} -- "$@"
+    command fzf-tmux ${FZF_TMUX_OPTS:--d${FZF_TMUX_HEIGHT:-40%}} -- "$@"
   else
     shift
-    fzf "$@"
+    command fzf "$@"
```

Comment:
I would allow users to use their versions of fzf and fzf-tmux, whether it's an alias or a function. By using a function, you can dynamically apply different sets of options depending on some conditions, e.g. light theme during daytime, dark theme at night.

Reply:
```suggestion
    fzf "$@"
```

Reply:
Hmm I was actually wondering about the same thing when I edited it, and had even planned to point that out and ask you for your opinion.

Thought now that you write it, I'm no longer sure whether it should really be using a potential alias/function.

1. In case if `fzf`/`fzf-tmux`, alias and function would most likely be used in the same way (if set by a user) - I mean to override the default behaviour. Whether via an alias or a function doesn't really matter, IMO.
2. Why would a user set an alias/function on either of those? I mean most likely to change some default behaviour, for which we in principle already have `FZF_DEFAULT_OPTS`. But that is used for both, `fzf`and `fzf-tmux`, ain’t it? So I could imagine a user might set up an alias/function to specifically target just one of these.
3. I'd also think, that users would rather not complete replace `fzf`/`fzf-tmux`, like with another program, or something that completely changes behaviour - I think it's more likely that they just set options to it.
4. Now, for the purpose of completion or key-binding functions... do we want to take such override effect?
  I mean what are people likely to set? Maybe things like `‐‐scheme=` or `--height` or so?
  But for those I'd rather say that we **don't** want them to take effect for the our completion or key-binding functions, do we?
  I rather think that **if** a user has such specific needs that wants him to override `fzf`/`fzf-tmux` in `__fzf_comprun()`, he should either override the function itself, or - if enough there was enough demand for that, we should add a specific env var that sets options for the case where this is used.

But of course, no choice is perfect. So up to you, what we do.

Reply:
As I said above, dynamic theming is a good example where users may feel $FZF_DEFAULT_OPTS isn't enough. After all, it doesn't matter what we think is a better way to do things. For a user who relies on an alias or a function, this is a breaking change. And we don't break user scripts unless there is a very compelling reason to do so.

Reply:
Rebased to current master, with the “protection” removed from `fzf`/`fzf-tmux`.

Please review & merge.